app/src/drawable/3d/label3d_handler.ts

```diff
@@ -120,13 +121,44 @@ export class Label3DHandler {
   }
 
   /**
-   * Handle double click, select label for editing
+   * Handle double click, select label for editing. Move ground plane if highlighted.
    *
+   * @param x
+   * @param y
    * @returns true if consumed, false otherwise
    */
-  public onDoubleClick(): boolean {
-    this.selectHighlighted()
-    return this._highlightedLabel !== null
+  public onDoubleClick(x: number, y: number): boolean {
+    const selectedLabel = Session.label3dList.selectedLabel
+    if (
+      this._highlightedLabel === null &&
+      selectLabel !== null &&
+      selectedLabel?.label.type === LabelTypeName.PLANE_3D
+    ) {
+      // Add to hist shapes for undo
+      const shape = Session.label3dList.getCurrentShape()
+      Session.label3dList.addShapeToHistShapes(shape)
```

Comment:
Was thinking if it might be a better UX to do FIFO-style (with a fixed-size buffer) for all shapes in the history (because the user loses the ability to undo if they accidentally select another shape)

Reply:
Good point! We can add that in a future PR, as we have to change multiple spots.